agent/connect/ca/provider_vault_test.go

```diff
@@ -51,6 +53,46 @@ func TestVaultCAProvider_SecondaryActiveIntermediate(t *testing.T) {
 	require.NoError(err)
 }
 
+func TestVaultCAProvider_RenewToken(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	skipIfVaultNotPresent(t)
+
+	testVault, err := runTestVault()
```

Comment:
Unless I'm missing it, it seems like this vault provider is never shut down. Either `runTestVault` or this test function should probably have:

```go
t.Cleanup(testVault.Stop)
```